services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -961,17 +962,27 @@ async def delete_project_simcore_s3(
 
         async with self.engine.acquire() as conn, conn.begin():
             # access layer
-            can: Optional[AccessRights] = await get_project_access_rights(
-                conn, int(user_id), project_id
-            )
-            if not can.delete:
-                logger.debug(
-                    "User %s was not allowed to delete project %s",
-                    user_id,
-                    project_id,
+            try:
+                can: Optional[AccessRights] = await get_project_access_rights(
+                    conn, int(user_id), project_id
                 )
-                raise web.HTTPForbidden(
-                    reason=f"User does not have delete access for {project_id}"
+                if not can.delete:
+                    logger.debug(
+                        "User %s was not allowed to delete project %s",
+                        user_id,
+                        project_id,
+                    )
+                    raise web.HTTPForbidden(
+                        reason=f"User does not have delete access for {project_id}"
+                    )
+            except ProjectNotFoundtError:
+                # When the project is deleted, first it is removed from the database
+                # and then the data is removed. If there is no project no access rights
+                # will be granted for the project to be remove.
+                # In this situation it is safe to remove the project
```

Comment:
yes, so for me this comment makes sense.

One question is, did you check the other functions that call get_project_access_rights ? and what happens if the project is not found?

Reply:
I had a look into it now. The error will propagate through the stack and eventually end up in the response to the client.

I guess that is fine, since this is already an edge case, and if the project is missing it is going to say that.